lib/exhal/collection.ex

```diff
@@ -14,7 +14,17 @@ defmodule ExHal.Collection do
       fn -> {:ok, a_doc} end,
       fn follow_result ->
         case follow_result do
-          {:error, _} -> {:halt, follow_result}
+          # End of Pagination
```

Comment:
Since this is now raising, we should probably document that behavior. Here's a proposal for an updated `@doc`:

```
@doc """
Returns a stream that iterates over the collection represented by `a_doc`.

In the case of failures during processing the stream, the stream will halt if 
an `ExHal.NoSuchLinkError` error occurs when following the next resource.  

If an `ExHal.Error` occurs when requesting the next resource, an 
`ExHal.CollectionError` will be raised to the caller.
"""
```

Reply:
Thanks for the suggestion. Updated.

Reply:
I think that second paragraph is a little misleading because the `NoSuchLinkError` is not a failure, it's our signal to stop iterating. Maybe add this to the first paragraph?

```
Iteration halts when there is no further `next` link to follow.
```

Reply:
That's a good point. I second that change.